### PR TITLE
alias: fetch peer certificate when adding an alias

### PR DESCRIPTION
**Pull request by aead**
This commit adds an extension to the `alias set`
command. When adding an alias fails due to a TLS
trust error - i.e. remote peer certificate verification
fails - `mc` fetches the remote peer certificate,
computes the public key fingerprint and asks the user
to confirm the fingerprint.

When the user confirms the fingerprint, the certificate
is added to the `CAs` directory.

This allows users to use a TOFU authentication with
self-signed certificates - similar to SSH.

### CODE REVIEW

File: cmd/alias-set.go [hunk 1]
```diff
@@ -184,45 +194,81 @@ func probeS3Signature(ctx context.Context, accessKey, secretKey, url string) (st
...
-		if _, err := s3Client.Stat(ctx, StatOptions{}); err != nil {
-			e := err.ToGoError()
-			if _, ok := e.(BucketDoesNotExist); ok {
-				// Bucket doesn't exist, means signature probing worked successfully.
-				return stype, nil
+		if strings.Contains(pErr.ToGoError().Error(), "x509: certificate signed by unknown authority") {
+			// We don't need to fetch the peer certificate
+			// if STDOUT is not a terminal and the user cannot
+			// confirm the public key.
+			if !term.IsTerminal(int(os.Stdout.Fd())) {
```
Comment (harshavardhana) [inline cmd/alias-set.go:216]: BTW we should also not ask for `globalIsJSON`  if set @aead

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/alias-set.go (hunk 1 above)
Comment (aead) [reply cmd/alias-set.go]: 👍

File: cmd/alias-set.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2015-2021 MinIO, Inc.
+// Copyright (c) 2013-2021 MinIO, Inc.
```
[the remark this answers is not in the document]
Comment (aead) [reply cmd/alias-set.go]: Not sure why this got changed... Reverted 👍

**Comment by vadmeste** [comment]
@aead just for simplificaton purpose, maybe it is better to do this outside probeS3Configuration(), this latter is doing too many things already

it can be done in mainAliasSet() before BuildS3Config(). I believe, you can just issue a GET request

**Comment by aead** [comment]
@vadmeste We can do that but that means `mc alias set` does one HTTP additional HTTP request in all cases.

**Review by harshavardhana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM & tested

**Pull request merged**